R/ParetoFiltering.R

```diff
@@ -52,21 +114,25 @@ plotPCNSE <- function(res, objectives = "r.squared", return.data = FALSE) {
     return(resm2)
   }
 
-  ggplot(resm2) +
-    geom_line(aes(x = sim.period, y = value, group = model, col = dominated, linetype = Model.str)) +
+  ggplot2::ggplot(resm2) +
+    ggplot2::geom_line(ggplot2::aes(x = sim.period, y = value, group = model, col = dominated, linetype = Model.str)) +
     ## geom_line(aes(x=sim.period,y=value,group=model),col="grey")+
-    geom_point(aes(x = sim.period, y = value, group = model, col = dominated)) +
-    geom_text(aes(x = sim.period, y = value, label = model),
+    ggplot2::geom_point(ggplot2::aes(x = sim.period, y = value, group = model, col = dominated)) +
+    ggplot2::geom_text(ggplot2::aes(x = sim.period, y = value, label = model),
       size = 3, hjust = -0.2,
       data = subset(resm2, sim.period == max(resm2$sim.period))
     ) +
-    geom_text(aes(x = sim.period, y = value, label = model),
+    ggplot2::geom_text(ggplot2::aes(x = sim.period, y = value, label = model),
       size = 3, hjust = 1.2,
       data = subset(resm2, sim.period == min(resm2$sim.period))
     ) +
-    scale_y_continuous(name = "Performance - NSE", breaks = seq(0.4, 1, by = 0.1)) +
-    scale_x_discrete(name = "Simulation period") +
-    scale_colour_discrete(name = "Dominated?") +
-    scale_linetype_discrete(name = "Model structure") +
-    coord_cartesian(ylim = c(0.4, 1)) ## FIXME
+    ggplot2::scale_y_continuous(name = "Performance - NSE", breaks = seq(0.4, 1, by = 0.1)) +
+    ggplot2::scale_x_discrete(name = "Simulation period") +
+    ggplot2::scale_colour_discrete(name = "Dominated?") +
+    ggplot2::scale_linetype_discrete(name = "Model structure") +
+    ggplot2::coord_cartesian(ylim = c(0.4, 1)) ## FIXME
 }
+
+
+#' @import utils
+utils::globalVariables(c("sim.period", "value", "model", "dominated", "Model.str"))
```

Comment:
I'm not familiar this. How did you work out which variables need to be declared as global?

Reply:
A lot of changes here were put in place to make R CMD check happy. It returned messages such as these (I know the below is from elsewhere, but I can't find the exact errors for this specific example at the moment):

```
paretoTimeAnalysis.crossvalidation: no visible binding for global
  variable ‘sim.period’
```

I followed advice on a couple of R websites on how to deal with this, and they suggested adding them as global variables with `util`.

[https://community.rstudio.com/t/how-to-solve-no-visible-binding-for-global-variable-note/28887/3](https://community.rstudio.com/t/how-to-solve-no-visible-binding-for-global-variable-note/28887/3)
[https://www.r-bloggers.com/no-visible-binding-for-global-variable/](https://www.r-bloggers.com/no-visible-binding-for-global-variable/)